src/main/kotlin/no/skatteetaten/aurora/mean/genie/service/SharedSecretReader.kt

```diff
@@ -0,0 +1,39 @@
+package no.skatteetaten.aurora.mean.genie.service
+
+import java.io.File
+import java.io.IOException
+import mu.KotlinLogging
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * Component for reading the shared secret used for authentication. You may specify the shared secret directly using
+ * the aurora.token.value property, or specify a file containing the secret with the aurora.token.location property.
+ */
+@Component
+class SharedSecretReader(
+    @Value("\${aurora.token.location:}") private val secretLocation: String?,
+    @Value("\${aurora.token.value:}") private val secretValue: String?
+) {
+
+    val secret = initSecret(secretValue)
```

Comment:
Hva er bakgrunnen for at secretValue sendes som parameter her alene, og ikke sammen med secretLocation? Evt. hvorfor leses ikke begge direkte som medlemsvariable istedenfor å sendes som parametre?

Dersom du sender begge som parametre trenger ikke secretLocation og secretValue være private val. Dersom du lar dem stå som private val trenger de ikke være parametre i det hele tatt, egentlig. Blir litt smakssak hva som er best, men akkurat nå mener jeg det er en mellomting. Dette kan jo fikses på i forbindelse med testingen, kanskje?

Reply:
Godt observert. Ser at dette også er tilfellet i Gobo. Jeg fjerner secretValue som parameter